superset/common/query_object.py

```diff
@@ -17,9 +17,9 @@ class QueryObject:
     def __init__(
             self,
             granularity: str,
-            groupby: List[str],
-            metrics: List[Metric],
-            filters: List[str],
+            groupby: List[str] = None,
```

Comment:
any reason to not just default these to `[]` instead of `None` since we do below anyway?

Reply:
In python, setting the default value as `[]` violates lint rules, and this is recommended way to do so.